### PR TITLE
[DOCS] Describe common use cases for functionbeat

### LINKED ISSUE #11867: Document Functionbeat use cases
We should better outline the Functionbeat use cases in the documentation for user clarity, either in the [overview page](https://www.elastic.co/guide/en/beats/functionbeat/current/functionbeat-overview.html) or a dedicated page.

Lambda is a natural integration point for many AWS services, and Functionbeat can be used in conjunction with it to stitch together integrations with many of these services. We currently have integrations with Cloudwatch Logs, SQS, and Kinesis, and we expect to enable more integrations over time. Currently, Functionbeat enables two core use cases:
1. Monitoring AWS services with Cloudwatch Logs
2. Event driven processing workflows with Kinesis and SQS

@dedemorton is this something you could help with? I'm happy to help provide more context if needed.

/cc @pieterhumphrey @DanRoscigno

### PR DESCRIPTION
**Pull request by dedemorton**
Closes #11867.

The text is ready to review, but I still need to add a couple diagrams, so I'm marking this as in progress.

TODO:

- [x] add diagrams.

@acchen97 I decided to break this content out into sections so that it will scale as we add support for more use cases that we want to highlight. I also think that adding sections helps users who are scanning the content. I struggled with the notion of presenting use cases without telling folks how to implement them. My compromise was to describe how the data passes through the parts of the stack. I plan to add diagrams so that folks who like to scan can see quickly which parts of the stack are relevant to the use case.

The svg diagram shown here is from the functionbeat page: https://www.elastic.co/products/beats/functionbeat

I'm trying to locate the source so I can use the same images/fonts in the diagrams that show the processing pipeline.

### CODE REVIEW

**Comment by dedemorton** [comment]
@acchen97 I've added a couple of diagrams. Not sure which ingest node processors make sense to show for these examples, though. Can you suggest a few to give a realistic pipeline example in the diagram?

File: x-pack/functionbeat/docs/overview.asciidoc [hunk 1]
```diff
@@ -10,12 +10,89 @@
...
+// REVIEWERS: I decided to talk about the underpinning technologies
+// here (like dissect processor, filtering, and ingest node) to help users
+// understand the flow of data. I've tried to put enough descriptive elements
+// to avoid confusing new users. I think some simple diagrams that show these
+// components will be useful to help users understand the parts involved.
+
+The processing pipeline for this use case typically looks like this:
+
+. {beatname_uc} runs as a Lambda function on AWS and reads the data stream from
+a Cloudwatch Log group.
```
Comment (acchen97) [inline x-pack/functionbeat/docs/overview.asciidoc]: @dedemorton FYI we're soon going to be enabled the use case here where Cloudwatch Logs can feed a Kinesis queue and Functionbeat will consume directly from Kinesis. This is important for organizations that may have multiple AWS subscriptions that all need to centralize their logs. We're still missing a few things to make this work such as gzip support: https://github.com/elastic/beats/pull/12119

Comment (acchen97) [reply x-pack/functionbeat/docs/overview.asciidoc]: Nevermind, should have kept reading. :)

File: x-pack/functionbeat/docs/overview.asciidoc [hunk 2]
```diff
@@ -10,12 +10,89 @@
...
+
+. {beats} processors, such as <<dissect,dissect>> and <<drop-fields,drop_fields>>,
+filter and structure the events.
+
+. Optional {ref}/ingest.html[ingest node] pipelines in {es} further enhance the
+data.
+
+. The structured events are indexed into an {es} cluster.
+
+image::./images/functionbeat-pipeline-cloudwatchlogs.png["{beatname_uc} collects events generated by CloudWatch logs"]
```
Comment (acchen97) [inline x-pack/functionbeat/docs/overview.asciidoc]: @ph @kvch thoughts on whether this use case makes sense for Cloudwatch logs processing?

File: x-pack/functionbeat/docs/overview.asciidoc [hunk 3]
```diff
@@ -10,12 +10,89 @@
...
+[float]
+[[event-drive-processing]]
+=== Perform event-driven processing
+
+You can use {beatname_uc} to implement event-driven processing workflows with
+cloud messaging queues and the {stack}. {beatname_uc} responds to event triggers
+from AWS Kinesis and SQS.
+
+[float]
+==== Analyze application data from SQS
```
Comment (acchen97) [inline x-pack/functionbeat/docs/overview.asciidoc]: I like this one. It's generic enough and shows a nice flow.

File: x-pack/functionbeat/docs/overview.asciidoc [hunk 4]
```diff
@@ -10,12 +10,89 @@
...
+filter and structure the events.
+
+. Optional {ref}/ingest.html[ingest node] pipelines in {es} further enhance the
+data.
+
+. The structured events are indexed into an {es} cluster.
+
+image::./images/functionbeat-pipeline-cloudwatchlogs.png["{beatname_uc} collects events generated by CloudWatch logs"]
+
+If you have multiple instances of CloudWatch logs and want to centralize your
```
Comment (acchen97) [inline x-pack/functionbeat/docs/overview.asciidoc]: Re: my comment above, should we omit this paragraph until we fully enable this architecture?

**Comment by acchen97** [comment]
Thanks @dedemorton. Shaping up nicely, left a few comments.

@kvch and @ph may have have more real-world processor/pipeline examples for these use cases.

File: x-pack/functionbeat/docs/overview.asciidoc (hunk 4 above)
Comment (dedemorton) [reply x-pack/functionbeat/docs/overview.asciidoc]: I can comment it out for now.

**Comment by dedemorton** [comment]
@kvch and @ph Do you have any details to provide here?

**Pull request merged**